packages/venia-ui/lib/components/Image/image.js

```diff
@@ -21,49 +29,79 @@ const Image = props => {
         onError,
         onLoad,
         placeholder,
+        resource,
+        resourceHeight,
+        resourceWidth,
+        sizes,
         src,
-        fileSrc,
+        type,
         ...rest
     } = props;
 
     const talonProps = useImage({
         onError,
-        onLoad,
-        placeholder
+        onLoad
     });
 
     const {
         handleError,
         handleImageLoad,
-        hasError,
         isLoaded,
         shouldRenderPlaceholder
     } = talonProps;
 
     const classes = mergeClasses(defaultClasses, propsClasses);
 
-    // Render a placeholder until the image is loaded.
-    const placeholderImage = shouldRenderPlaceholder ? (
-        <img className={classes.root} src={placeholder} alt={alt} {...rest} />
-    ) : null;
+    // A placeholder to use until the image is loaded.
+    const placeholderClass = shouldRenderPlaceholder
+        ? classes.placeholder_include
+        : classes.placeholder_exclude;
+    const placeholderImage = (
+        <img
+            alt={alt}
+            className={placeholderClass}
+            loading="eager"
+            src={placeholder}
+            {...rest}
+        />
+    );
+
+    /*
+     * These don't live in the talon because they depend on @magento/venia-drivers.
+     */
+    const imageSrcset = useMemo(() => generateSrcset(resource, type), [
+        resource,
+        type
+    ]);
+    const source = useMemo(() => {
+        // If we have a direct src, use it.
+        // Otherwise, get a resourceUrl from the resource and use that.
+        return src
+            ? src
+            : resourceUrl(resource, {
+                  type,
+                  height: resourceHeight,
+                  width: resourceWidth
+              });
+    }, [resource, resourceHeight, resourceWidth, src, type]);
 
     const imageClass =
         classes.root + ' ' + (isLoaded ? classes.loaded : classes.notLoaded);
-
-    const imageSrcset = useMemo(
-        () => generateSrcset(fileSrc, 'image-product'),
-        [fileSrc]
-    );
-
-    const actualImage = !hasError && (
+    const actualImage = (
+        /*
+         * Note: attributes that are allowed to be overridden
+         * must appear before the spread of `rest`.
+         */
         <img
+            loading="lazy"
             {...rest}
             alt={alt}
             className={imageClass}
             onError={handleError}
             onLoad={handleImageLoad}
-            src={src}
+            src={source}
             srcSet={imageSrcset}
+            sizes={sizes}
```

Comment:
Good feedback, I did the same for `width` and `height` too. I'm not really a fan of the spread props for things like this though because then the API isn't clear.

I suppose I could add them to `propTypes` but not destructure them.

Reply:
Ok I went to do this but only the actual image needs `sizes` (not the placeholder), so I kept the destructure so that I could spread `rest` out onto both the actual image and the placeholder.